test/annotation_utils.t

```diff
@@ -864,6 +873,170 @@ subtest '_run_rast_workflow_ann' => sub {
+                      "$ftr->{location}[0][0]";
+                if( $cnt > $cnt_limit ) {
+                    last;
+                }
+            }
+        }
+        if( $pgn->{non_coding_features} &&
+            @{$pgn->{non_coding_features}} > 0) {
+            $cnt = 0;
+            for my $nc_ftr (@{$pgn->{non_coding_features}}) {
+                $cnt += 1;
+                print "\n------the rasted genome nc_feature location's contig_id=".
+                      "$nc_ftr->{location}[0][0]";
+                if( $cnt > $cnt_limit ) {
+                    last;
+                }
+            }
+        }
+        if( $pgn->{cdss} && @{$pgn->{cdss}} > 0 ) {
+            $cnt = 0;
+            for my $cds (@{$pgn->{cdss}}) {
+                $cnt += 1;
+                print "\n------the rasted genome cdss location's contig_id=".
+                      "$cds->{location}[0][0]";
+                if( $cnt > $cnt_limit ) {
+                    last;
+                }
+            }
+        }
+        if( $pgn->{mrnas} && @{$pgn->{mrnas}} > 0 ) {
+            $cnt = 0;
+            for my $mrna (@{$pgn->{mrnas}}) {
+                $cnt += 1;
+                print "\n------the rasted genome mrnas location's contig_id=".
+                      "$mrna->{location}[0][0]";
+                if( $cnt > $cnt_limit ) {
+                    last;
+                }
+            }
+        }
+        my $ctgID_hash = $rd{contigID_hash};
+        if( $ctgID_hash ) {
+            print "\nThere is a contigID_hash:\n".Dumper($ctgID_hash);
+        }
+        else {
+            print "\nThere is NO contigID_hash in $pgn->{id}:\n".Dumper(keys %rd);
+        }
+        ## Run the remapping
+        my $pgn1 = $annoutil->_remap_contigIDs( $ctgID_hash, $pgn );
+        print "\n=========after remapping the rasted genome on $pgn->{id}:\n";
+
+        if( $pgn1->{contig_ids} &&  @{$pgn1->{contig_ids}} > 0) {
+            print "\n------the rasted genome's contig_ids:\n".
+                   Dumper( $pgn1->{contig_ids});
+            my $exp_ctg_ids = $pgn->{contig_ids};
+            for my $cid (@{$exp_ctg_ids}) {
+                $cid = $ctgID_hash->{$cid};
+            }
+            cmp_deeply $pgn1->{contig_ids}, $exp_ctg_ids,
+                        'contig ids remapped correctly';
+        }
+        if( $pgn1->{contigs} &&  @{$pgn1->{contigs}} > 0) {
+            for( my $i = 0; $i < @{$pgn1->{contigs}}; $i++ ) {
+                my $ctg_id = @{$pgn1->{contigs}}[$i]->{id};
+                my $old_ctg_id = @{$pgn->{contigs}}[$i]->{id};
+                if( $i < $cnt_limit ) {
+                  print "\n------the rasted genome contig's id=$ctg_id";
+                }
+                ok( (!defined($ctgID_hash->{$old_ctg_id}) && $ctg_id eq $old_ctg_id) ||
+                     $ctg_id eq $ctgID_hash->{$old_ctg_id},
+                    'id of contigs re-mapped correctly' );
+            }
+        }
```

Comment:
could you replace this with a `cmp_deeply` test instead?

Reply:
changed